R/save_arcadia_plot.R

```diff
@@ -0,0 +1,62 @@
+#' Save Arcadia Plot
+#'
+#' @param plot ggplot object to be saved
+#' @param filepath The path where the plot will be saved
+#' @param panel_size The size of the panel, one of "full_wide", "float_wide", "half_square", "full_square", "float_square"
+#' @param transparent_bg Logical, whether to set the plot background to be transparent
+#'
+#' @return None
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' p <- ggplot(mtcars, aes(x = hp, y = mpg)) + geom_point()
+#' save_arcadia_plot(p, "plot.pdf", "full_wide")
+#' }
+save_arcadia_plot <- function(filepath, plot, panel_size = c("full_wide", "float_wide", "half_square", "full_square", "float_square")) {
```

Comment:
To avoid duplication, you might pull out the list of panel size names into a constant, and unpack the list as needed later (e.g. when erroring on invalid panel size).